src/App.jsx

```diff
@@ -14,33 +16,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ABC = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPrepereGoods(goods, { sortField, reversed }) {
+  let prepareGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField === SORT_FIELD_ABC) {
+    prepareGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField === SORT_FIELD_LENGTH) {
+    prepareGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (reversed) {
+    prepareGoods = prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortedField] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getPrepereGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortedField(SORT_FIELD_ABC)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_FIELD_ABC,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortedField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReversed(prev => !prev)}
+          type="button"
+          className={cn('button is-warning', { 'is-light': !reversed })}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reversed) && (
+          <button
+            onClick={() => {
+              setSortedField('');
+              setReversed(false);
+            }}
```

Comment:
Move reset logic to the helper variable and use it here
```suggestion
            onClick={reset}
```